### PR TITLE
Formally deprecate distro.linux_distribution()

### PR DESCRIPTION
**Pull request by jdufresne**
Users should use distro.id(), distro.version(), and distro.name()
instead.

Refs #263

### CODE REVIEW

**Comment by HorlogeSkynet** [comment]
There is a typo : s/compatability/compatibility/.

**Review by sethmlarson — COMMENTED** [review]
Looks good, one nit:

**Comment by jdufresne** [comment]
Thanks for the reviews. I addressed all feedback in the latest revision.

**Review by sethmlarson — APPROVED** [review]
LGTM, pinging @nir0s as this is a big manifesto change. After this should we remove mentions to `distro.linux_distribution()` in the README, repository description, etc

**Comment by HorlogeSkynet** [comment]
Agreeing with @sethmlarson here. From my PoV this project (first) intended to be a drop-in replacement for the `platform` deprecated method. Maybe the official Python documentation should be updated too, as it points to this module for this specific purpose ?

**Comment by jdufresne** [comment]
I'll hold off on merging until @nir0s has a chance to weigh in.

**Comment by nir0s** [comment]
Guys, I appreciate you holding off, but let's not make a habit out of it. If you feel like there's a huge change, and you want me to pitch in, I'll try. However, if you think that it makes sense to do something, and I'm unresponsive, go ahead and do it.

Anyway, to address the comments:

> LGTM, pinging @nir0s as this is a big manifesto change. After this should we remove mentions to distro.linux_distribution() in the README, repository description, etc

Indeed we should.

Agreeing 

> with @sethmlarson here. From my PoV this project (first) intended to be a drop-in replacement for the platform deprecated method. Maybe the official Python documentation should be updated too, as it points to this module for this specific purpose?

It's only my opinion, but I see no reason why there's a need for a single function that returns a tuple with an unclear structure. What's the first part? Is it the full name? The same questions apply to the codename and the version. Given that explicit is better than implicit, I'd rather do something like:

```python
if distro.id() == "..." and distro.name() == "...":
   ...
```

And not

I indeed think that the official Python documentation should be updated.

**Comment by nir0s** [comment]
Guys, I appreciate you holding off, but let's not make a habit out of it. If you feel like there's a huge change, and you want me to pitch in, I'll try. However, if you think that it makes sense to do something, and I'm unresponsive, go ahead and do it.

Anyway, to address the comments:

> LGTM, pinging @nir0s as this is a big manifesto change. After this should we remove mentions to distro.linux_distribution() in the README, repository description, etc

Indeed we should.

> with @sethmlarson here. From my PoV this project (first) intended to be a drop-in replacement for the platform deprecated method. Maybe the official Python documentation should be updated too, as it points to this module for this specific purpose?

It's only my opinion, but I see no reason why there's a need for a single function that returns a tuple with an unclear structure. What's the first part? Is it the full name? The same questions apply to the codename and the version. Given that explicit is better than implicit, I'd rather do something like:

```python
if distro.id() == "..." and distro.name() == "...":
   ...
```

which also allows me to pass fields explicitly to the relevant function (e.g. `best`),

Instead of:

```python
distro_info = distro.linux_distribution()

if distro_info[0] == "you get the point":
  ...
```

I indeed think that the official Python documentation should be updated, and I don't mind deprecating this until distro v2.0 comes out, if only to give people a shitload of time to stop using it.

Honestly, I'm not strongly opinionated about this. `distro` is supposed to provide a reasonable alternative to an annoying problem. If the Python community responds badly to removing `linux_distribution`, AFAIC, we can leave it be. It's not like it takes a lot of effort to maintain.

However, even if we do decide to keep it, we can push to update the official Python docs to encourage people to use `distro.id()`, etc.. instead.

Thoughts?

**Comment by jdufresne** [comment]
Thanks. 

> I indeed think that the official Python documentation should be updated, and I don't mind deprecating this until distro v2.0 comes out, if only to give people a shitload of time to stop using it.

This makes sense. The warning is a good start and there is no rush to remove or anything like that.

**Comment by HorlogeSkynet** [comment]
@python-distro/maintainers 

Dear maintainers, I guess we should update the GitHub repository description now that this deprecation has been released.

Any idea ? What about ?

```diff
- A much more elaborate, renewed alternative implementation for Python's platform.linux_distribution()
+ A much more elaborate replacement for old Python's `platform.linux_distribution()` method
```

Thanks 🙏

**Pull request merged**
